src/components/pages/SetPassword/SetPassword.jsx

```diff
@@ -0,0 +1,156 @@
+import React, { useCallback, Fragment, useState } from 'react'
+import { Field, Form } from 'react-final-form'
+
+import PasswordField from 'components/layout/form/fields/PasswordField'
+import { LegalInfos } from 'components/layout/LegalInfos/LegalInfos'
+import Logo from 'components/layout/Logo'
+import PageTitle from 'components/layout/PageTitle/PageTitle'
+import { redirectLoggedUser } from 'components/router/helpers'
+import { setPassword } from 'repository/pcapi/pcapi'
+
+export const TOKEN_ERROR_MESSAGE = 'Votre lien de création de mot de passe est invalide'
+export const PASSWORD_CREATED_MESSAGE = 'Votre mot de passe a bien été créé.'
+export const INVALID_FORM_MESSAGE = "Une erreur s'est produite, veuillez corriger le formulaire."
+export const UNKNOWN_ERROR_MESSAGE = "Une erreur s'est produite, veuillez contacter le support."
+export const DIFFERENT_PASSWORDS_ERROR_MESSAGE = 'Les deux mots de passe ne sont pas identiques'
+
+export const SetPassword = props => {
+  const { currentUser, history, match, showNotification } = props
+
+  redirectLoggedUser(history, currentUser)
+
+  const redirectOnTokenError = useCallback(() => {
+    showNotification('error', 'Votre lien de création de mot de passe est invalide')
+    history.push('/mot-de-passe-perdu')
+  }, [showNotification, history])
+
+  const token = match?.params.token
+
+  if (!token) {
+    redirectOnTokenError()
+  }
+
+  const [backendErrors, setBackendErrors] = useState(null)
+
+  const submitSetFirstPassword = useCallback(
+    values =>
+      setPassword(token, values.password)
+        .then(() => {
+          showNotification('success', PASSWORD_CREATED_MESSAGE)
+          history.push('/connexion')
+        })
+        .catch(error => {
+          if (error.errors.newPassword) {
+            showNotification('error', INVALID_FORM_MESSAGE)
+            setBackendErrors(error.errors)
+            return
+          }
+          if (error.errors.token) {
+            redirectOnTokenError()
+            return
+          }
+          showNotification('error', UNKNOWN_ERROR_MESSAGE)
+        }),
+    [showNotification, history, token, redirectOnTokenError]
+  )
+
+  const getPasswordErrors = useCallback(
+    errors => {
+      if (errors?.password && errors.password !== true) {
+        return errors.password
+      }
+
+      if (backendErrors && backendErrors.newPassword) {
```

Comment:
Tu peux utiliser la même logique qu'au-dessus pour simplifier

```suggestion
      if (backendErrors?.newPassword) {
```